File: HashTable/twoSum.py
#https://leetcode.com/problems/two-sum/
class Solution(object):
    def twoSum(self, nums, target):
        d = {}
        for i in range(0, len(nums)):
            if (target - nums[i]) in d:
                return [ d[target - nums[i]], i]
            else:
                d[ nums[i] ] = i
        
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        

File: JavaScript/twoSum.js
// https://leetcode.com/problems/two-sum/
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    var visited = {}
    for (var i = 0; i < nums.length; i++) {
        difference = target - nums[i]
        if (difference in visited) {
            return [i, visited[difference]];
        }
        visited[nums[i]] = i;
    }
    return [-1, -1];
};